core/sdk/src/utils/ws_utils.rs

```diff
@@ -220,20 +219,20 @@ impl WebSocketUtils {
                     }
                 }
             }
-
-            if (index % LOG_RETRY_INTERVAL).eq(&0) {
-                error!(
-                    "Websocket TCP Connection with {} failed with retry for last {} secs in {}",
-                    url_path, index, tcp_port
-                );
-            }
             if let Some(fail) = &config.fail_after {
                 if fail.eq(&index) {
                     break Err(RippleError::NotAvailable);
                 }
             }
             index += 1;
             tokio::time::sleep(Duration::from_millis(retry_every)).await;
+            let duration = Duration::from_millis(retry_every);
```

Comment:
What is the initial value `retry_every`.   You may set the initial value of retry_every to 50 milliseconds, and use an exponential backoff strategy, doubling the delay on each retry, with a maximum delay capped at 3 seconds.